### PR TITLE
feat(geography): add geography testing module to mock facade usage

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/geography/src/facades/geography/geography-facade.interface.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Observable } from 'rxjs';
+import { Action } from '@ngrx/store';
+import { Dictionary } from '@ngrx/entity';
+
+import { DaffCountry } from '../../models/country';
+
+export interface DaffGeographyFacadeInterface<T extends DaffCountry = DaffCountry> {
```
Comment (griest024) [inline libs/geography/src/facades/geography/geography-facade.interface.ts:7]: Extend from `DaffStoreFacade<Action>` here.

File: libs/geography/src/facades/geography/geography.facade.ts [hunk 2]
```diff
@@ -10,11 +10,12 @@ import {
   getDaffGeographySelectors
 } from '../../selectors/public_api';
 import { DaffCountry } from '../../models/country';
+import { DaffGeographyFacadeInterface } from './geography-facade.interface';
 
 @Injectable({
   providedIn: 'root'
 })
-export class DaffGeographyFacade<T extends DaffCountry> implements DaffStoreFacade<Action> {
+export class DaffGeographyFacade<T extends DaffCountry> implements DaffStoreFacade<Action>, DaffGeographyFacadeInterface<T> {
```
Comment (griest024) [inline libs/geography/src/facades/geography/geography.facade.ts:18]: Don't implement `DaffStoreFacade<Action>` here.

File: libs/geography/src/facades/geography/geography-facade.interface.ts (hunk 1 above)
Comment (lderrickable) [reply libs/geography/src/facades/geography/geography-facade.interface.ts]: done

File: libs/geography/src/facades/geography/geography.facade.ts (hunk 2 above)
Comment (lderrickable) [reply libs/geography/src/facades/geography/geography.facade.ts]: done

**Pull request merged**